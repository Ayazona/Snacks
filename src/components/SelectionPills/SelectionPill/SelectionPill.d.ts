import * as React from 'react'
import { WithThemeInjectedProps, ApplyWithTheme } from '../../../styles/themer/withTheme'
import { RadiumStyles, ElementAttributes } from '../../..'

export interface SelectionPillProps extends WithThemeInjectedProps {
  /** Required unique identifier for the pill */
  id: string | number

  /** Any additional props to add to the list element (e.g. data attributes). */
  listElementAttributes?: ElementAttributes<'li'>

  /** Any additional props to add to the checkbox element (e.g. data attributes). */
  elementAttributes?: ElementAttributes<'Checkbox'>

  /** Determines wether or not selected styles are applied and start is a selected state */
  isSelected?: boolean

  /** Determines Whether or not the pill is disabled. */
  isDisabled?: boolean

  /** Callback function called after pill click */
  onClick(): void

  /** Callback function called after pill gained focus */
  onFocus(): void

  /** Callback function called after pill has lost focus */
  onBlur(): void

  /** Text to appear inside pill */
  text?: string

  /** Optional styles. */
  style?: RadiumStyles

  /** Aria overrides for accessibility (i.e. use if label is not descriptive enough for screen readers) */
  aria?: { label ?: string }
}

declare const SelectionPillProps: ApplyWithTheme<React.ComponentType<SelectionPillProps>>

export default SelectionPillProps
