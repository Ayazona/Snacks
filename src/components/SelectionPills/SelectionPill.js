import React from 'react'
import PropTypes from 'prop-types'
import Radium from 'radium'
import colors from '../../styles/colors'
import withTheme from '../../styles/themer/withTheme'
import Checkbox from '../Buttons/Checkbox'
import { themePropTypes } from '../../styles/themer/utils'
import spacing from '../../styles/spacing'
import { setAlpha } from '../../utils'

const STYLES = {
  container: {
    display: 'inline-block',
  },
  checkbox: {
    display: 'block',
    fontWeight: 600,
    fontSize: 14,
    height: 32,
    lineHeight: '32px',
    borderRadius: 24,
    borderWidth: 1,
    borderStyle: 'solid',
    borderColor: colors.GRAY_88,
    marginTop: 0,
    marginBottom: 0,
    marginLeft: '4px',
    marginRight: '4px',
    ...spacing.PADDING_X_SM,
    transition: 'background-color 150ms ease-in-out',
  },
  checkBoxOverrideStyle: {
    border: 0,
    clip: 'rect(0 0 0 0)',
    clippath: 'inset(50%)',
    margin: -1,
    overflow: 'hidden',
    padding: 0,
    position: 'absolute',
    whiteSpace: 'nowrap',
    height: 1,
    width: 1,
  },
}

class SelectionPill extends React.PureComponent {
  static propTypes = {
    /** Required unique identifier for the pill */
    id: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,

    /** Any additional props to add to the list element (e.g. data attributes). */
    listElementAttributes: PropTypes.object,

    /** Any additional props to add to the checkbox element (e.g. data attributes). */
    elementAttributes: PropTypes.object,

    /** Determines wether or not selected styles are applied and start is a selected state */
    isSelected: PropTypes.bool,

    /** Determines Whether or not the pill is disabled. */
    isDisabled: PropTypes.bool,

    /** Callback function called after pill click
     * @param {SyntheticEvent} event The react `SyntheticEvent`
     * @param {props} object All the props passed to the component
     */
    onClick: PropTypes.func,

    /** Callback function called after pill gained focus
     * @param {SyntheticEvent} event The react `SyntheticEvent`
     * @param {props} object All the props passed to the component
     */
    onFocus: PropTypes.func,

    /** Callback function called after pill has lost focus
      * @param {SyntheticEvent} event The react `SyntheticEvent`
      * @param {props} object All the props passed to the component
      */
    onBlur: PropTypes.func,

    snacksTheme: themePropTypes,
    /** Text to appear inside pill */
    text: PropTypes.string,

    /** Optional style overrides. */
    style: PropTypes.object,

    /** Aria overrides for accessibility (i.e. use if label is not descriptive enough for screen readers) */
    aria: PropTypes.shape({
      label: PropTypes.string,
    }),
  }

  static defaultProps = {
    elementAttributes: {},
    isSelected: false,
  }

  state = {
    isSelected: this.props.isSelected,
    isFocused: false,
  }

  handleChange = event => {
    const { onClick, isDisabled } = this.props
    const { isSelected } = this.state
    if (isDisabled) return

    this.setState({ isSelected: !isSelected })
    onClick(event, { ...this.props, isSelected: !isSelected })
  }

  toggleFocus = () => {
    const { isFocused } = this.state
    this.setState({ isFocused: !isFocused })
  }

  handleFocus = event => {
    const { onFocus, isDisabled } = this.props
    const { isFocused } = this.state
    if (isDisabled) return

    this.toggleFocus()
    onFocus(event, { ...this.props, isFocused: !isFocused })
  }

  handleBlur = event => {
    const { onBlur, isDisabled } = this.props
    const { isFocused } = this.state
    if (isDisabled) return

    this.toggleFocus()
    onBlur(event, { ...this.props, isFocused: !isFocused })
  }

  render() {
    const { id, snacksTheme, text, style, aria, isDisabled } = this.props
    const { isSelected, isFocused } = this.state
    const { primaryForeground } = snacksTheme.colors
    const backgroundColor = setAlpha(primaryForeground, 0.1)

    const activeStyles = {
      backgroundColor,
      borderColor: primaryForeground,
      ':hover': {
        backgroundColor: setAlpha(primaryForeground, 0.3),
      },
    }

    const checkboxThemedStyles = {
      color: primaryForeground,
      ':hover': {
        backgroundColor,
      },
    }

    const disabledStyles = {
      color: setAlpha(primaryForeground, 0.5),
      backgroundColor: setAlpha(colors.GRAY_88, 0.5),
      ':hover': {
        backgroundColor: setAlpha(colors.GRAY_88, 0.5),
      },
    }

    const focusStyles = {
      outline: '5px auto rgb(59, 153, 252)',
      borderColor: primaryForeground,
    }

    return (
      <li style={STYLES.container} {...this.props.listElementAttributes}>
        <Checkbox
          id={id}
          onChange={this.handleChange}
          onFocus={this.handleFocus}
          onBlur={this.handleBlur}
          aria={aria}
          style={{
            label: {
              ...STYLES.checkbox,
              ...checkboxThemedStyles,
              ...(isSelected && activeStyles),
              ...(isFocused && focusStyles),
              ...(isDisabled && disabledStyles),
              ...style,
            },
            button: STYLES.checkBoxOverrideStyle,
          }}
          {...this.props.elementAttributes}
      >
          {text}
        </Checkbox>
      </li>
    )
  }
}

export default withTheme(Radium(SelectionPill))
