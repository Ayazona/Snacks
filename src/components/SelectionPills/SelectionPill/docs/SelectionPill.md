### tl;dr
`SelectionPill` is a checkbox based component used for multi-selection purposes, such as filters.
This differs from `NavigationPill`, which supports single selection in the form of a link.
The main distinction between these components is the accessibility support, which will
inform screen reader users how to interact with the pill appropriately.

**`SelectionPill` example:**

`SelectionPill` maintains its own internal state for selected and focused, but
can also be controlled by it's parent when passing in the `parentControlledState`
prop. When this prop is present, internal state is ignored and the `isSelected`
controls whether a `SelectionPill` is is selected or not.

```jsx
import { SelectionPill } from 'ic-snacks'

; <div>
    <SelectionPill
      isSelected={true}
      onClick={e => {
        console.log('SelectionPill click!', "carrots")
      }}
      text="carrots"
    />
    <SelectionPill
      isSelected={false}
      onClick={e => {
        console.log('SelectionPill click!', "watermelon")
      }}
      text="watermelon"
    />
    <SelectionPill
      id="selection3"
      isSelected={false}
      isDisabled={true}
      onClick={e => {
        console.log('SelectionPill click!', "cabbage")
      }}
      text="cabbage"
    />
  </div>
```
