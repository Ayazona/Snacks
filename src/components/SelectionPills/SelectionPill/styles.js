import colors from '../../../styles/colors'
import { setAlpha } from '../../../utils'
import spacing from '../../../styles/spacing'

export const getStyles = (props, primaryForeground) => {
  const backgroundColor = setAlpha(primaryForeground, 0.1)

  return {
    container: {
      display: 'inline-block',
    },
    checkbox: {
      display: 'block',
      fontWeight: 600,
      fontSize: 14,
      height: 32,
      lineHeight: '32px',
      borderRadius: 24,
      borderWidth: 1,
      borderStyle: 'solid',
      borderColor: colors.GRAY_88,
      color: primaryForeground,
      marginTop: 0,
      marginBottom: 0,
      marginLeft: '4px',
      marginRight: '4px',
      ...spacing.PADDING_X_SM,
      transition: 'background-color 150ms ease-in-out',
      ':hover': {
        backgroundColor,
      },
    },
    checkBoxOverrideStyle: {
      border: 0,
      clip: 'rect(0 0 0 0)',
      clippath: 'inset(50%)',
      margin: -1,
      overflow: 'hidden',
      padding: 0,
      position: 'absolute',
      whiteSpace: 'nowrap',
      height: 1,
      width: 1,
    },
    focusStyles: {
      outline: '5px auto rgb(59, 153, 252)',
      borderColor: primaryForeground,
    },
    disabledStyles: {
      color: setAlpha(primaryForeground, 0.6),
      backgroundColor: colors.GRAY_97,
      ':hover': {
        backgroundColor: colors.GRAY_97,
      },
    },
    activeStyles: {
      backgroundColor,
      borderColor: primaryForeground,
      ':hover': {
        backgroundColor: setAlpha(primaryForeground, 0.3),
      },
    },
  }
}
