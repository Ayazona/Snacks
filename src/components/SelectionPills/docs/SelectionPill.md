SelectionPill example:

```jsx
// import { SelectionPill } from 'ic-snacks'
import SelectionPill from '../SelectionPill.js'


; <div>
    <SelectionPill
      id="selection1"
      isSelected={false}
      onClick={e => {
        console.log('SelectionPill click!', "bananas")
      }}
      text="bananas"
    />
    <SelectionPill
      id="selection2"
      isSelected={true}
      onClick={e => {
        console.log('SelectionPill click!', "carrots")
      }}
      text="carrots"
    />
    <SelectionPill
      id="selection3"
      isSelected={false}
      onClick={e => {
        console.log('SelectionPill click!', "watermelon")
      }}
      text="watermelon"
    />
  </div>
```
