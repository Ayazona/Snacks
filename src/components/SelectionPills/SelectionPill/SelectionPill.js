import React from 'react'
import Radium from 'radium'
import PropTypes from 'prop-types'
import { themePropTypes } from '../../../styles/themer/utils'
import withTheme from '../../../styles/themer/withTheme'
import { getStyles } from './styles'

const NoOp = () => {} // eslint-disable-line no-empty-function
class SelectionPill extends React.PureComponent {
  static propTypes = {
    /** Required unique identifier for the pill */
    id: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,

    /** Any additional props to add to the list element (e.g. data attributes). */
    listElementAttributes: PropTypes.object,

    /** Any additional props to add to the checkbox element (e.g. data attributes). */
    elementAttributes: PropTypes.object,

    /** Flag determining if component selected state is controlled by parent through props or internal state */
    parentControlledState: PropTypes.bool,

    /** Determines wether or not selected styles are applied and start is a selected state */
    isSelected: PropTypes.bool,

    /** Determines Whether or not the pill is disabled. */
    isDisabled: PropTypes.bool,

    /** Callback function called after pill click
     * @param {SyntheticEvent} event The react `SyntheticEvent`
     * @param {props} object All the props passed to the component
     */
    onClick: PropTypes.func,

    /** Callback function called after pill gained focus
     * @param {SyntheticEvent} event The react `SyntheticEvent`
     * @param {props} object All the props passed to the component
     */
    onFocus: PropTypes.func,

    /** Callback function called after pill has lost focus
     * @param {SyntheticEvent} event The react `SyntheticEvent`
     * @param {props} object All the props passed to the component
     */
    onBlur: PropTypes.func,

    /** Snacks theme attributes provided by Themer */
    snacksTheme: themePropTypes,

    /** Text to appear inside pill */
    text: PropTypes.string,

    /** Optional style overrides for button and its states */
    style: PropTypes.shape({
      button: PropTypes.object,
      disabledStyle: PropTypes.object,
      selectedStyle: PropTypes.object,
      focusedStyle: PropTypes.object,
    }),

    /** Aria overrides for accessibility (i.e. use if label is not descriptive enough for screen readers) */
    aria: PropTypes.shape({
      label: PropTypes.string,
    }),
  }

  static defaultProps = {
    elementAttributes: {},
    isSelected: false,
    parentControlledState: false,
    onClick: NoOp,
    onFocus: NoOp,
    onBlur: NoOp,
    style: {},
    aria: {},
  }

  state = {
    isSelected: this.props.isSelected,
    isFocused: false,
  }

  handleChange = event => {
    const { onClick } = this.props
    const { isSelected } = this.state

    this.setState({ isSelected: !isSelected })
    onClick(event, { ...this.props, isSelected: !isSelected })
  }

  toggleFocus = () => {
    const { isFocused } = this.state
    this.setState({ isFocused: !isFocused })
  }

  handleFocus = event => {
    this.toggleFocus()
    this.props.onFocus(event, { ...this.props, isFocused: !this.state.isFocused })
  }

  handleBlur = event => {
    this.toggleFocus()
    this.props.onBlur(event, { ...this.props, isFocused: !this.state.isFocused })
  }

  renderInputBtn({ selected }) {
    const { aria, isDisabled, id } = this.props
    const componentStyles = getStyles({})

    return (
      <div style={componentStyles.checkBoxOverrideStyle}>
        <input
          id={id}
          type="checkbox"
          onChange={this.handleChange}
          checked={selected}
          disabled={isDisabled}
          aria-label={aria.label}
          onBlur={this.handleBlur}
          onFocus={this.handleFocus}
        />
      </div>
    )
  }

  render() {
    const { id, snacksTheme, text, style, isDisabled, parentControlledState } = this.props
    const { isFocused } = this.state
    const { primaryForeground } = snacksTheme.colors
    const selected = parentControlledState ? this.props.isSelected : this.state.isSelected
    const componentStyles = getStyles({
      isSelected: selected,
      isFocused,
      isDisabled,
      primaryForeground,
      externalStyles: style,
    })

    return (
      <li style={componentStyles.listElement} {...this.props.listElementAttributes}>
        {this.renderInputBtn({ selected })}
        <label htmlFor={id} style={componentStyles.labelButton}>
          {text}
        </label>
      </li>
    )
  }
}

export default withTheme(Radium(SelectionPill))
