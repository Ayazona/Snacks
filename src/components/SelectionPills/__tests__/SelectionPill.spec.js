import React from 'react'
import renderer from 'react-test-renderer'
import SelectionPill from '../SelectionPill'

it('renders selected pill correctly', () => {
  const tree = renderer
    .create(<SelectionPill id="selection1" isSelected text={'selected pill'} />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders unselected pill correctly', () => {
  const tree = renderer
    .create(<SelectionPill id="selection1" isSelected={false} text={'Not selected pill'} />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders disabled pill correctly', () => {
  const tree = renderer
    .create(<SelectionPill id="selection1" isDisabled text={'Disabled pill'} />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders element attributes on pill correctly', () => {
  const tree = renderer
    .create(
      <SelectionPill
        id="selection1"
        text={'pill with attributes'}
        listElementAttributes={{ style: { margin: '10px' } }}
      />
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders element attributes on pill correctly', () => {
  const tree = renderer
    .create(
      <SelectionPill
        id="selection1"
        text={'pill with attributes'}
        listElementAttributes={{ style: { margin: '10px' } }}
        elementAttributes={{ role: 'tab' }}
      />
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
