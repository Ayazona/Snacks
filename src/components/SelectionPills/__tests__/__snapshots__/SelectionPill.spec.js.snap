// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders disabled pill correctly 1`] = `
<li
  data-radium={true}
  style={
    Object {
      "display": "inline-block",
    }
  }
>
  <div
    data-radium={true}
    style={
      Object {
        "display": "flex",
      }
    }
  >
    <div
      data-radium={true}
      style={
        Object {
          "border": 0,
          "clip": "rect(0 0 0 0)",
          "clippath": "inset(50%)",
          "height": 1,
          "margin": -1,
          "overflow": "hidden",
          "padding": 0,
          "position": "absolute",
          "whiteSpace": "nowrap",
          "width": 1,
        }
      }
    >
      <test-file-stub
        aria-hidden="true"
        data-radium={true}
        style={
          Object {
            "fill": "#757575",
            "height": 22,
            "width": 22,
          }
        }
      />
      <input
        checked={false}
        data-radium={true}
        id="selection1"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        style={
          Object {
            "MozAppearance": "none",
            "MsAppearance": "none",
            "WebkitAppearance": "none",
            "appearance": "none",
            "backgroundImage": "none",
            "cursor": "pointer",
            "height": 22,
            "left": 0,
            "opacity": 0,
            "position": "absolute",
            "top": 0,
            "width": 22,
          }
        }
        type="checkbox"
      />
    </div>
    <label
      data-radium={true}
      htmlFor="selection1"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={
        Object {
          "backgroundColor": "#F7F7F7",
          "borderColor": "#E0E0E0",
          "borderRadius": 24,
          "borderStyle": "solid",
          "borderWidth": 1,
          "color": "rgba(67, 176, 42, 0.6)",
          "display": "block",
          "fontSize": 14,
          "fontWeight": 600,
          "height": 32,
          "lineHeight": "32px",
          "marginBottom": 0,
          "marginLeft": "4px",
          "marginRight": "4px",
          "marginTop": 0,
          "paddingLeft": 16,
          "paddingRight": 16,
          "transition": "background-color 150ms ease-in-out",
        }
      }
    >
      Disabled pill
    </label>
  </div>
</li>
`;

exports[`renders element attributes on pill correctly 1`] = `
<li
  data-radium={true}
  style={
    Object {
      "margin": "10px",
    }
  }
>
  <div
    data-radium={true}
    style={
      Object {
        "display": "flex",
      }
    }
  >
    <div
      data-radium={true}
      style={
        Object {
          "border": 0,
          "clip": "rect(0 0 0 0)",
          "clippath": "inset(50%)",
          "height": 1,
          "margin": -1,
          "overflow": "hidden",
          "padding": 0,
          "position": "absolute",
          "whiteSpace": "nowrap",
          "width": 1,
        }
      }
    >
      <test-file-stub
        aria-hidden="true"
        data-radium={true}
        style={
          Object {
            "fill": "#757575",
            "height": 22,
            "width": 22,
          }
        }
      />
      <input
        checked={false}
        data-radium={true}
        id="selection1"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        style={
          Object {
            "MozAppearance": "none",
            "MsAppearance": "none",
            "WebkitAppearance": "none",
            "appearance": "none",
            "backgroundImage": "none",
            "cursor": "pointer",
            "height": 22,
            "left": 0,
            "opacity": 0,
            "position": "absolute",
            "top": 0,
            "width": 22,
          }
        }
        type="checkbox"
      />
    </div>
    <label
      data-radium={true}
      htmlFor="selection1"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={
        Object {
          "borderColor": "#E0E0E0",
          "borderRadius": 24,
          "borderStyle": "solid",
          "borderWidth": 1,
          "color": "#43B02A",
          "display": "block",
          "fontSize": 14,
          "fontWeight": 600,
          "height": 32,
          "lineHeight": "32px",
          "marginBottom": 0,
          "marginLeft": "4px",
          "marginRight": "4px",
          "marginTop": 0,
          "paddingLeft": 16,
          "paddingRight": 16,
          "transition": "background-color 150ms ease-in-out",
        }
      }
    >
      pill with attributes
    </label>
  </div>
</li>
`;

exports[`renders element attributes on pill correctly 2`] = `
<li
  data-radium={true}
  style={
    Object {
      "margin": "10px",
    }
  }
>
  <div
    data-radium={true}
    style={
      Object {
        "display": "flex",
      }
    }
  >
    <div
      data-radium={true}
      style={
        Object {
          "border": 0,
          "clip": "rect(0 0 0 0)",
          "clippath": "inset(50%)",
          "height": 1,
          "margin": -1,
          "overflow": "hidden",
          "padding": 0,
          "position": "absolute",
          "whiteSpace": "nowrap",
          "width": 1,
        }
      }
    >
      <test-file-stub
        aria-hidden="true"
        data-radium={true}
        style={
          Object {
            "fill": "#757575",
            "height": 22,
            "width": 22,
          }
        }
      />
      <input
        checked={false}
        data-radium={true}
        id="selection1"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        style={
          Object {
            "MozAppearance": "none",
            "MsAppearance": "none",
            "WebkitAppearance": "none",
            "appearance": "none",
            "backgroundImage": "none",
            "cursor": "pointer",
            "height": 22,
            "left": 0,
            "opacity": 0,
            "position": "absolute",
            "top": 0,
            "width": 22,
          }
        }
        type="checkbox"
      />
    </div>
    <label
      data-radium={true}
      htmlFor="selection1"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={
        Object {
          "borderColor": "#E0E0E0",
          "borderRadius": 24,
          "borderStyle": "solid",
          "borderWidth": 1,
          "color": "#43B02A",
          "display": "block",
          "fontSize": 14,
          "fontWeight": 600,
          "height": 32,
          "lineHeight": "32px",
          "marginBottom": 0,
          "marginLeft": "4px",
          "marginRight": "4px",
          "marginTop": 0,
          "paddingLeft": 16,
          "paddingRight": 16,
          "transition": "background-color 150ms ease-in-out",
        }
      }
    >
      pill with attributes
    </label>
  </div>
</li>
`;

exports[`renders selected pill correctly 1`] = `
<li
  data-radium={true}
  style={
    Object {
      "display": "inline-block",
    }
  }
>
  <div
    data-radium={true}
    style={
      Object {
        "display": "flex",
      }
    }
  >
    <div
      data-radium={true}
      style={
        Object {
          "border": 0,
          "clip": "rect(0 0 0 0)",
          "clippath": "inset(50%)",
          "height": 1,
          "margin": -1,
          "overflow": "hidden",
          "padding": 0,
          "position": "absolute",
          "whiteSpace": "nowrap",
          "width": 1,
        }
      }
    >
      <test-file-stub
        aria-hidden="true"
        data-radium={true}
        style={
          Object {
            "fill": "#757575",
            "height": 22,
            "width": 22,
          }
        }
      />
      <input
        checked={false}
        data-radium={true}
        id="selection1"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        style={
          Object {
            "MozAppearance": "none",
            "MsAppearance": "none",
            "WebkitAppearance": "none",
            "appearance": "none",
            "backgroundImage": "none",
            "cursor": "pointer",
            "height": 22,
            "left": 0,
            "opacity": 0,
            "position": "absolute",
            "top": 0,
            "width": 22,
          }
        }
        type="checkbox"
      />
    </div>
    <label
      data-radium={true}
      htmlFor="selection1"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={
        Object {
          "backgroundColor": "rgba(67, 176, 42, 0.1)",
          "borderColor": "#43B02A",
          "borderRadius": 24,
          "borderStyle": "solid",
          "borderWidth": 1,
          "color": "#43B02A",
          "display": "block",
          "fontSize": 14,
          "fontWeight": 600,
          "height": 32,
          "lineHeight": "32px",
          "marginBottom": 0,
          "marginLeft": "4px",
          "marginRight": "4px",
          "marginTop": 0,
          "paddingLeft": 16,
          "paddingRight": 16,
          "transition": "background-color 150ms ease-in-out",
        }
      }
    >
      selected pill
    </label>
  </div>
</li>
`;

exports[`renders unselected pill correctly 1`] = `
<li
  data-radium={true}
  style={
    Object {
      "display": "inline-block",
    }
  }
>
  <div
    data-radium={true}
    style={
      Object {
        "display": "flex",
      }
    }
  >
    <div
      data-radium={true}
      style={
        Object {
          "border": 0,
          "clip": "rect(0 0 0 0)",
          "clippath": "inset(50%)",
          "height": 1,
          "margin": -1,
          "overflow": "hidden",
          "padding": 0,
          "position": "absolute",
          "whiteSpace": "nowrap",
          "width": 1,
        }
      }
    >
      <test-file-stub
        aria-hidden="true"
        data-radium={true}
        style={
          Object {
            "fill": "#757575",
            "height": 22,
            "width": 22,
          }
        }
      />
      <input
        checked={false}
        data-radium={true}
        id="selection1"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        style={
          Object {
            "MozAppearance": "none",
            "MsAppearance": "none",
            "WebkitAppearance": "none",
            "appearance": "none",
            "backgroundImage": "none",
            "cursor": "pointer",
            "height": 22,
            "left": 0,
            "opacity": 0,
            "position": "absolute",
            "top": 0,
            "width": 22,
          }
        }
        type="checkbox"
      />
    </div>
    <label
      data-radium={true}
      htmlFor="selection1"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={
        Object {
          "borderColor": "#E0E0E0",
          "borderRadius": 24,
          "borderStyle": "solid",
          "borderWidth": 1,
          "color": "#43B02A",
          "display": "block",
          "fontSize": 14,
          "fontWeight": 600,
          "height": 32,
          "lineHeight": "32px",
          "marginBottom": 0,
          "marginLeft": "4px",
          "marginRight": "4px",
          "marginTop": 0,
          "paddingLeft": 16,
          "paddingRight": 16,
          "transition": "background-color 150ms ease-in-out",
        }
      }
    >
      Not selected pill
    </label>
  </div>
</li>
`;
